projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/base/LLVMNode.java

```diff
@@ -48,4 +50,28 @@
 
     public abstract void executeVoid(VirtualFrame frame);
 
+    protected boolean isLLVMAddress(Object object) {
+        return object instanceof LLVMAddress;
+    }
+
+    @SuppressWarnings("unused")
+    protected boolean isUninitialized(VirtualFrame frame, LLVMGlobalVariableStorage globalVariableStorage) {
```

Comment:
I'm not sure if it is a good idea to have `isUninitialized`, `isInitialized`, `isNative` and `isManaged` in the `LLVMNode` class, since the methods are not really related to it. I think one of the two options would work better:
1. Directly call the guard, which would create a direct correspondence between the guard (e.g., `globalVariableStorage.isInitialized`) and initialize method (e.g., `globalVariableStorage.initialize`).
2. Make a helper class for it, if you think that it is likely that we will replace the guard test in the future.